### ISSUE TITLE
btrfs-restore sometimes waits for user input

### ISSUE THREAD

**Issue by zoqaeski** [danthem/undelete-btrfs#5]
I noticed while trying to restore some huge files with _lots_ of extents that `btrfs restore` would sometimes wait for user input if it looped too many times while searching the filesystem tree (or whatever it does). By sending all of `btrfs restore`s output to `/dev/null`, this feedback is lost and the script apparently hangs—I had to terminate a run because the CPU usage of `btrfs restore` had sat at 0 for about half an hour.

**Comment by danthem** [comment]
Do you have a copy of what the prompt says? I've never come across it on my test setups but I've never really needed to recover large files with a lot of extents.. I'll see if I can repro and figure out a nice way around it.

**Comment by zoqaeski** [comment]
Not off the top of my head. It's something like "`We seem to be looping on FILE at EXTENT too many times. Continue? y/N/a`". It's unclear as to whether it blocks indefinitely, or gives up after a timeout; I only terminated your script once when I was doing a restore on `/Music/S.*/`—the second time it worked without a hitch. (I was playing it safe by restoring each letter of the alphabet individually, because I wasn't sure how many files got deleted.)

I've noticed from a dry run of `rsync` that DVD VOB files and videos of concerts (>150 MB in size) are the only truncated files (though thankfully these weren't among the files deleted). The files were also quite old, and possibly fragmented? I'd never done a defragment on the drive.

There might be more information here? https://github.com/kdave/btrfs-progs/issues/59
